crates/bevy_render/src/render_asset.rs

```diff
@@ -180,16 +183,23 @@ fn extract_render_asset<A: RenderAsset>(
         }
     }
 
-    let mut extracted_assets = Vec::new();
+    let mut extracted_assets = std::mem::take(&mut extracted_assets.extracted_overflow);
     for id in changed_assets.drain() {
```

Comment:
What happens if an `AssetEvent::Added {x}` or `AssetEvent::Modified {x}` gets fired on two consecutive frames, such that the asset with id `x` overflows on the first frame?

If I'm reading this correctly, the outcome would be that asset with id `x` gets extracted twice on the second frame.

Similarly if an asset overflows and on the next frame gets `AssetEvent::Removed {x}` then it will both be extracted and put on the `removed` pile.

I don't know much about bevy rendering, so I can't tell if what I'm saying is relevant, but it struck me as potentially unexpected behaviour, which wouldn't have happened previous to this PR (due to the HashSet  `changed_assets`) so other code *might* be subtly depending on it not happening.
However, this code seems to be consistent with the previous one in that it eventually processes all the events it normally would.

Unrelated and idk if this belongs in a comment on this PR, but this system has a typo in the docstring `crated` should be `created`.
Also, should the docstring be extended with a mention of this overflowing behaviour?

Reply:
Actually on a second reading I realized that an event being both extracted and put on the `removed` pile was possible, provided one emits an `AssetEvent::Added { id } | AssetEvent::Modified { id }` event and an `AssetEvent::Removed { id }` in such a way that the removal event arrives before the other one. Again, not sure if this is intended/completely fine, but it makes this system dependent on event-order, which doesn't seem like a good thing to me. (But that's maybe not a problem, and is perhaps beyond the scope of this PR?)